src/main/scala/com/fulcrumgenomics/util/UpdateDelimitedFileContigNames.scala

```diff
@@ -99,17 +117,81 @@ class UpdateDelimitedFileContigNames
           case Some(column) =>
             throw new IllegalStateException(s"Did not find contig ${fields(column)} in the given sequence dictionary.")
           case None =>
-            // Write it all out
-            forloop(from = 0, until = numFields) { i =>
-              if (0 < i) out.append(delimiter)
-              out.write(fields(i))
+            // sorting
+            sorter match {
+              case None => // Write out the fields without sorting
+                forloop(from = 0, until = numFields) { i =>
+                  if (0 < i) out.append(delimiter)
+                  out.write(fields(i))
+                }
+                out.write('\n')
+              case Some(_sorter) => // add then to the sorter, to be written later
+                _sorter += CoordinateAndString(
+                  contig   = dict(fields(contigColumnIndex)).index,
+                  position = if (this.sortOrder == SortOrder.ByContigOnly) 0 else position.map(i => fields(i).toInt).getOrElse(0),
+                  lineno   = progress.getCount.toInt,
+                  line     = fields.take(numFields).mkString(delimiterString)
+                )
             }
-            out.write('\n')
         }
       }
     }
     progress.logLast()
 
+    // Sort and write the output
+    sorter.foreach { _sorter =>
+      logger.info("Sorting the output")
+      _sorter.foreach { data =>
+        out.write(data.line)
+        out.write('\n')
+      }
+      _sorter.close()
+    }
+
     out.close()
   }
+}
+
+/** Trait that entries in SortOrder will extend. */
+sealed trait SortOrder extends EnumEntry
```

Comment:
Could all of this stuff go into a companion object please?

Reply:
Unfortunately `FgBioEnum`s cannot be in the companion object, but the rest can.